### PR TITLE
Add tests and code improvements for notus

### PR DESCRIPTION
**Pull request by jjnicola**
**What**:
Add tests for Notus metadata uploader and some code improvements. 
The class accept the Nvti Cache as optional argument.

**Checklist**:

- [x] Tests
- [x] [CHANGELOG](https://github.com/greenbone/gsa/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

**Comment by jjnicola** [comment]
@thorsten-passfeld Maybe you want to take a look.

File: tests/notus/example.csv [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# General metadata:
```
Comment (thorsten-passfeld) [inline tests/notus/example.csv:1]: Note that this and the corresponding test will have to be updated at least once more after certain small changes to the CSV format.

**Comment by thorsten-passfeld** [comment]
Looks good to me! The unit tests are good.

File: ospd_openvas/notus/metadata.py [hunk 2]
```diff
@@ -268,94 +395,22 @@ def update_metadata(self) -> None:
...
-                    )
-                    # BIDs
-                    advisory_metadata_list.append("")
-                    # XREFS
-                    advisory_metadata_list.append(
-                        self._format_xrefs(
-                            advisory_dict["ADVISORY_XREF"],
-                            ast.literal_eval(advisory_dict["XREFS"]),
-                        )
+                file_name = PurePath(csv_file.name).name
```
Comment (bjoernricks) [inline ospd_openvas/notus/metadata.py:408]: Is there any reason for using a PurePath here directly?

File: tests/notus/test_notus.py [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+from collections import OrderedDict
+from unittest.mock import patch, MagicMock
+
+from ospd_openvas.notus.metadata import (
+    NotusMetadataHandler,
+    EXPECTED_FIELD_NAMES_LIST,
+    METADATA_DIRECTORY_NAME,
+)
+from ospd_openvas.errors import OspdOpenvasError
+from tests.helper import assert_called_once, assert_called
```
Comment (bjoernricks) [inline tests/notus/test_notus.py:33]: I am not sure if these helpers are still necessary if we are using Python >= 3.7 now :-)

**Review by bjoernricks — APPROVED** [review]

**Comment by jjnicola** [comment]
Thanks @bjoernricks !

**Pull request merged**